packages/website/docs/templates.md

```diff
@@ -0,0 +1,221 @@
+---
+id: templates
+title: Displaying items with Templates
+---
+
+Templates let you customize the display of your autocomplete's items.
+
+Once you've set up your [data sources](sources), you need to define how they display in your autocomplete experience. It encompasses the structure for each item and the way they look.
+
+Autocomplete provides a Templates API to let you fully customize the render of each item.
+
+## Usage
+
+### Rendering each item
+
+The rendering system of Autocomplete uses an agnostic virtual DOM implementation. You can return anything from each template as long as they're valid virtual DOM elements (VNodes).
+
+For example, templates can return a string:
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return `Result: ${item.name}`;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+Or a [Preact](https://preactjs.com/) component:
+
+```js
+/** @jsx h */
+import { h } from 'preact';
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item }) {
+            return <div>{item.name}</div>;
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+### Returning HTML
+
+Native HTML elements aren't valid VNodes, which means you can't return a template string that contains HTML, or an HTML element. But if you're not using a virtual DOM implementation in your app, there are still two ways you can return HTML.
+
+#### Using `dangerouslySetInnerHTML`
+
+Each template function provides access to `createElement` and `Fragment` to create VNodes.
+
+```js
+import { autocomplete } from '@algolia/autocomplete-js';
+
+autocomplete({
+  // ...
+  getSources() {
+    return [
+      {
+        // ...
+        templates: {
+          item({ item, createElement, Fragment }) {
+            return createElement(Fragment, {}, item.name);
+          },
+        },
+      },
+    ];
+  },
+});
+```
+
+By default, `createElement` and `Fragment` default to [Preact](https://preactjs.com/)'s `Preact.createElement` (or `h`) and `preact.Fragment`. You can customize these and provide the virtual DOM implementation you prefer.
```

Comment:
I didn't write anything about how to provide your own implementation of `createElement` and `Fragment` because it seems like an advanced use case (plugging the library to a UI library with a custom VDOM implementation by hand). I'm not sure it belongs to "core" concepts.

Reply:
Right—we will write an advanced guide on this at some point.